### ISSUE TITLE
Const items only used in global_asm! are marked unused

### ISSUE THREAD

**Issue by Aiden2207** [rust-lang/rust#90335]
Consider the following code:

```rust
#![feature(global_asm)]
const FOO: u32 = 42;
global_asm! {
    ".bar = {foo}",
    foo = const FOO
}
```
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=dcfdd49de87ad756be000dd8021419bd)

Here `FOO` is only used to pass a value into the `global_asm` macro.  Rustc incorrectly reports that `FOO` is unused despite the fact that `FOO` is used. The regular `asm` macro correctly does not issue a warning in the [equivalent code.](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=5fb119783165ffa9a976b8ec8bc31056)

**Comment by fmease** [comment]
Updated:

```rs
#![feature(asm_const)]
const FOO: u32 = 42;
std::arch::global_asm! {
    ".bar = {foo}",
    foo = const FOO
}
```

This no longer triggers `dead_code`. Closing as fixed.
